config/application.rb

```diff
@@ -18,5 +20,7 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+
   end
+  UnknownError = Class.new(StandardError)
```

Comment:
this is new to me -- is it just a way to flag "we shouldn't get here" ?   If so, why not have an actual error message, which could be enough with StandardError?  Ditto where used below.

Reply:
This is a class declaration, not an object declaration or a `raise`.

Reply:
ah - I misread what was above and thought it was being returned by a method.  

So essentially, we have inherited this from the original code ... and my thinking about its use is smack on, as this is the result of grepping:

```
[ndushay@li-dl-7064-18cd pre-assembly (master)]$ ack Unknown
lib/pre_assembly/digital_object.rb
142:        raise PreAssembly::UnknownError unless result.class == String
347:        raise PreAssembly::UnknownError unless result && [200, 201, 202, 204].include?(result.code)

lib/pre_assembly.rb
42:  UnknownError = Class.new(StandardError)
```

ah well, more imperfections to ignore.